### PR TITLE
Add extra checks to `Graph` and more comments

### PR DESCRIPTION
**Pull request by pmnoxx**
We will add extra checks to make sure `RoutingTable` updates `Graph` properly. In addition I added documentation for `Graph`.

### CODE REVIEW

File: chain/network/src/routing.rs [hunk 1]
```diff
@@ -876,23 +876,34 @@ pub struct RoutingTableInfo {
     pub peer_forwarding: HashMap<PeerId, Vec<PeerId>>,
 }
 
+/// `Graph` is used to compute `peer_routing`, which contains information how to route messages to
```
Comment (mm-near) [inline chain/network/src/routing.rs:879]: Thank you for adding documentation, it helps a lot.

File: chain/network/src/routing.rs [hunk 2]
```diff
@@ -876,23 +876,34 @@ pub struct RoutingTableInfo {
...
+/// all known peers. That is, for each `peer`, we get a sub-set of peers to which we are connected
+/// to that are on the shortest path between us as destination `peer`.
 #[derive(Clone)]
 pub struct Graph {
-    pub source: PeerId,
+    /// peer_id of current peer
+    my_peer_id: PeerId,
+    /// `id` as integer corresponding to `my_peer_id`
```
Comment (mm-near) [inline chain/network/src/routing.rs:886]: can you also mention why do we need the u32 ids at all? is it for performance reasons?

File: chain/network/src/routing.rs [hunk 3]
```diff
@@ -876,23 +876,34 @@ pub struct RoutingTableInfo {
...
     p2id: HashMap<PeerId, u32>,
+    /// List of existing `PeerId`s
     id2p: Vec<PeerId>,
-    pub used: Vec<bool>,
-    pub unused: Vec<u32>,
+    /// Which ids are currently in use
+    used: Vec<bool>,
+    /// List of unused peer ids
+    unused: Vec<u32>,
+    /// Compressed adjacency table, we use 32 bit integer as ids instead of using full `PeerId`.
```
Comment (mm-near) [inline chain/network/src/routing.rs:896]: btw - is our graph directed or undirected?
do we store edges in both directions?

File: chain/network/src/routing.rs [hunk 4]
```diff
@@ -909,6 +920,14 @@ impl Graph {
         res
     }
 
+    pub fn total_active_edges(&self) -> u64 {
+        self.total_active_edges
+    }
+
+    pub fn compute_total_active_edges(&self) -> u64 {
```
Comment (mm-near) [inline chain/network/src/routing.rs:927]: naming is a little bit confusing -- as 'total_active_edges' returns X, while 'compute_total_active_edges' returns 2*X..

**Review by mm-near — APPROVED** [review]

File: chain/network/src/routing.rs (hunk 2 above)
Comment (pmnoxx) [reply chain/network/src/routing.rs:886]: Yes, I rewrote this code in order to use u32 to improve performance.

File: chain/network/src/routing.rs (hunk 3 above)
Comment (pmnoxx) [reply chain/network/src/routing.rs:896]: Yes, that's an undirected graph, we store edges in both directions.

File: chain/network/src/routing.rs (hunk 4 above)
Comment (pmnoxx) [reply chain/network/src/routing.rs:930]: Ok, I simplified the code to return the same value.

**Pull request merged**